axiom/userbase.py

```diff
@@ -283,8 +334,12 @@ def setPassword(self, newPassword):
 
         @return: A deferred firing when the password has been changed.
         """
-        self.password = None if newPassword is None else unicode(newPassword)
-        return succeed(None)
+        realm = self.store.findUnique(LoginSystem)
```

Comment:
If `setPassword` or `replacePassword` is called on the user-store `LoginAccount` item, this `findUnique` query will fail for obvious reasons.

```
>>> from twisted.cred.portal import IRealm
>>> from axiom.userbase import LoginSystem, LoginAccount
>>> IRealm(db).accountByAddress(u'jonathan', u'localhost').avatars.open().findUnique(LoginAccount)
LoginAccount(avatars=reference(-1), disabled=0, password=None, passwordHash=u'$argon2i$v=19$m=102400,t=2,p=8$jpHSmnMOoXSOsVaqtRaiVA$g/nVGDUMYCdSpoLmL27rcw', storeID=1)@0x110d25610
>>> _.store.findUnique(LoginSystem)
Traceback (most recent call last):
  File "<axiom browser>", line 1, in <module>
  File "/Users/jonathan/.virtualenvs/axiom/lib/python2.7/site-packages/axiom/store.py", line 1631, in findUnique
    raise errors.ItemNotFound(comparison)
ItemNotFound: None
```